### DISCUSSION TITLE
填空宏三个参数顺序改进

### DISCUSSION THREAD

**Discussion by dongrentianyu** [oflg/Tidme#19] [Ideas]
正常来说只需要用到一个参数就可以了。但在多个填空题的情况下，通常就需要用到id参数。而我常见的做法就是把答案复制一遍，这样id就不容易重名。但这个过程中，展示内容的参数就放在中间了。

所以调整一下三个参数的顺序是否会更好？

当然我可以自己diy改进一下，只是提个想法。

另一个解决办法就是，如果id参数能只在本条目里生效，而不影响其他条目，也是可以的。因为id就可以直接用123命名了。也不会影响到其他条目。但这样反而更难实现。

---

感觉提issue不太好，就关闭放到这边来讨论了。

**Comment by oflg** [comment]
我用的过程中也发现了这个问题，挺容易改这个代码改，只是改了后和原来的就不兼容了。

目前可以这样`<<C  空  id:1  >>`

**Reply by oflg** [reply]
@dongrentianyu 改顺序后，会与之前的版本不兼容。如果你能接受，我就这样改哈。

**Reply by dongrentianyu** [reply]
嗯，如果能改顺序的话，那还是很好的。因为多个填空题的卡片我现在还比较少。

先是挖空的内容，后是id，再是隐藏时展示出来的提示。感觉这样的顺序会比较好一点。主要是编写起来方便。因为隐藏时的展示好像不太管。当然也可能有的人会需要，作为一种提示，比如<<C  北京 城市名 id:1  >>这样子，通过城市名来提示要挖空的内容。

所以你先在群里调查一下，因为我个人的需求并不等于普遍的需求。虽然我个人觉得这种提示没啥用，我也没咋用到。

**Comment by dongrentianyu** [comment]
主要还是担心id重名了，虽然重名了好像也不太会影响正常复习。只要上下两个id不是连着的就可以了。

**Reply by oflg** [reply]
id重名是指条目A有id1，下一个条目B也有id1，然后两个A会影响B对吗？

**Comment by oflg** [comment]
> 如果id参数能只在本条目里生效，而不影响其他条目

我不太明白这句话。在设计id时就是只在本条目生效的，如果影响了其他条目，应该是出现了bug。

即：如果复习条目A时，条目A的id1的填空被点击展示后，复习下一个条目B时，条目b的相同id1的填空，设计上是不展示（没有被点击过）的。

**Reply by oflg** [reply]
如果是这个bug我就尽快修哈。

**Reply by dongrentianyu** [reply]
我刚才测试了，id确实会影响到其他条目，在复习状态下。不在复习状态下，也会影响其他条目，如果里面的填空设置的id相同的话。

也就是有两个id相同的，然后一个点击展开，另一个也会展开。当然我是特意把两个条目都摆出来了。正常复习的时候，好像只会展示一个条目，所以才显得不影响。

**Comment by oflg** [comment]
> id确实会影响到其他条目

确实。两个条目同时展示是会这样，我后面修下这个bug吧。

**Comment by oflg** [comment]
@dongrentianyu 刚刚发布了[v1.17.1](https://github.com/oflg/Tidme/releases/tag/v1.17.1)（需要tw核心v5.3.2及以上），已经修改了填空题条目的id互相影响的bug。也把参数顺序改了。

![image](https://github.com/oflg/Tidme/assets/42492105/db086c69-665f-4233-a691-91ab50a4704a)

**Reply by dongrentianyu** [reply]
这有点难搞了。太记因为某些特殊原因，还停留在5.3.0的预发版上。林一二说是核心去掉了一个什么功能，会影响react插件，接着就会影响依赖react插件的插件。所以太记一直没有更新版本。

唉，我再等等吧。核心就算要把这个功能加回去，也只能是下一个版本实现了。
